### PR TITLE
fix: refetching tx history after clearing tx history

### PR DESCRIPTION
**Pull request by theoboldfrazier**
## Description

When tx provider was rerendering, it was clearing the tx history without refetching from the rest endpoint.

## Notice

- [ ] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Risk

Minimal risk. Refetching a lot right now, but that is a separate issue with tx provider re-rendering whenever an asset page is loaded.

### CODE REVIEW

File: src/context/TransactionsProvider/TransactionsProvider.tsx [hunk 1]
```diff
@@ -83,6 +71,15 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
               },
               (err: any) => console.error(err)
             )
+            if (isEmpty(accountSpecifiers)) return
```
Comment (0xdef1cafe) [inline src/context/TransactionsProvider/TransactionsProvider.tsx:74]:
```suggestion
            if (isEmpty(accountSpecifiers)) continue
```

Comment (theoboldfrazier) [reply src/context/TransactionsProvider/TransactionsProvider.tsx]: Good catch. Done.

**Review by gomesalexandre — APPROVED** [review]

File: src/context/TransactionsProvider/TransactionsProvider.tsx [hunk 2]
```diff
@@ -83,6 +71,15 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
               },
               (err: any) => console.error(err)
             )
+            if (isEmpty(accountSpecifiers)) continue
+            accountSpecifiers.forEach(accountSpecifierMap => {
+              dispatch(
+                txHistoryApi.endpoints.getAllTxHistory.initiate(
+                  { accountSpecifierMap },
```
Comment (0xdef1cafe) [inline src/context/TransactionsProvider/TransactionsProvider.tsx:78]: see the logic in transactions provider of how we narrow this full `accountSpecifiers` array down to the individual map that's required, we don't have to fetch all of these restfully yet

Comment (theoboldfrazier) [reply src/context/TransactionsProvider/TransactionsProvider.tsx]: Reworked the call to move outside the inner loop and now getting account specifiers by caip2 and fetching tx history based on the caip2's account specifiers.

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**